ops/terraform/services/common/app-autoscaling.tf

```diff
@@ -1,6 +1,15 @@
-resource "aws_appautoscaling_target" "replicas" {
-  # only applicable for test environment
-  count = local.env == "test" ? 1 : 0
+locals {
+  enable_rds_scheduled_scaling = !var.disable_rds_scheduling_override && (local.env == "test" || local.is_ephemeral_env)
+  replicas_scaling_target      = local.enable_rds_scheduled_scaling ? one(aws_appautoscaling_target.dynamic_replicas) : one(aws_appautoscaling_target.static_replicas)
+}
+
+resource "aws_appautoscaling_target" "dynamic_replicas" {
```

Comment:
Note that we _do_ need two variants of this `aws_appautoscaling_target` resource specifically because `lifecycle` meta-arguments only support [literal values](https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#literal-values-only). So, we can't do something like:

```terraform
lifecycle {
  ignore_changes = local.enable_rds_scheduled_scaling ? [
    min_capacity,
    max_capacity
  ] : []
}
```

This is because the ternary cannot be evaluated at the time where the `lifecycle` meta-argument is evaluated, so Terrraform doesn't allow it. Instead, we have two variants of this resource: `dynamic_replicas` for when scheduled scaling of RDS nodes is enabled and `static_replicas` for when it isn't. Which variant is chosen and used by dependent resources depends on the `local.enable_rds_scheduled_scaling` (see the definition for `local.replicas_scaling_target`).